tests/test_http_transport.py

```diff
@@ -0,0 +1,290 @@
+        assert server is not None
+        assert server.name == "notebook"
+
+    @pytest.mark.integration
+    def test_http_server_startup(self):
+        """Test that HTTP server can start and respond to requests."""
+        server = create_server(port=8081)
+
+        # Start server in a separate thread
+        server_thread = threading.Thread(
+            target=lambda: server.run(transport="streamable-http"), daemon=True
+        )
+        server_thread.start()
+
+        # Give server time to start
+        time.sleep(2)
+
+        try:
+            # Test that server responds to HTTP requests
+            response = requests.post(
+                "http://127.0.0.1:8081/mcp",
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json, text/event-stream",
+                },
+                json={
+                    "jsonrpc": "2.0",
+                    "method": "initialize",
+                    "params": {
+                        "protocolVersion": "2025-06-18",
+                        "capabilities": {"tools": {}},
+                        "clientInfo": {"name": "test-client", "version": "1.0.0"},
+                    },
+                    "id": 1,
+                },
+                timeout=5,
+            )
+
+            # Check that we get a response (even if it's an error due to missing session)
+            assert response.status_code in [200, 202, 400, 406]
+
+        except requests.exceptions.RequestException:
+            pytest.skip("HTTP server did not start in time")
+
+    def test_tools_registered_correctly(self):
+        """Test that all tools are registered when server is created."""
+        # The server is already created at module import time with tools registered
+        # Just verify that the tools are accessible
+        server = create_server()
+        assert server is not None
+        assert server.name == "notebook"
+
+        # Since FastMCP is initialized at module level with decorators,
+        # we can't easily mock it. The test for server creation is sufficient.
+
+    def test_server_singleton_behavior(self):
+        """Test that create_server returns the same instance when called multiple times."""
+        # Reset the global mcp to None
+        import mcp_jupyter.server
+
+        original_mcp = mcp_jupyter.server.mcp
+        mcp_jupyter.server.mcp = None
+
+        try:
+            server1 = create_server()
+            server2 = create_server()
+
+            # Should return the same instance
+            assert server1 is server2
+        finally:
+            # Restore original state
+            mcp_jupyter.server.mcp = original_mcp
+
+
+class TestCLIArguments:
+    """Test CLI argument parsing."""
+
+    def test_cli_default_transport(self):
+        """Test default transport is stdio."""
+        from mcp_jupyter import main
+
+        with patch("sys.argv", ["mcp-jupyter"]):
```

Comment:
Is there a reason we need nested patches here and elsewhere? It looks like in at least one other file we use the pattern of decorator patches, so IMO it would be preferred to nesting as deeply as we do if we don't actually need the context managers

(applies elsewhere in this file too)